activesupport/lib/active_support/core_ext/securerandom.rb

```diff
@@ -41,7 +41,7 @@ def self.base58(n = 16)
   #
   #   p SecureRandom.base36 # => "4kugl2pdqmscqtje"
   #   p SecureRandom.base36(24) # => "77tmhrhjfvfdwodq8w7ev2m7"
-  if RUBY_VERSION >= "3.3"
+  if RUBY_VERSION >= "3.3" && Gem::Version.new(SecureRandom::VERSION) >= Gem::Version.new("0.3")
```

Comment:
```suggestion
  if SecureRandom::VERSION >= "0.3"
```

Checking both the Ruby version and gem version is redundant, only the gem version matter.

Reply:
I'm not familiar with the Ruby side of things, but if I'm using securerandom 3.1 with Ruby 2.7, for example, will this work well?

- https://github.com/ruby/ruby/pull/8312

Reply:
Sorry, forget my above comment. The above question came to me because I wasn't sure of the causal relationship between SecureRandom and the implementation on Ruby's side, but after reading the source code, it wasn't particularly relevant 😇

Reply:
No worries, I patched your PR. 

CI seems to be failing but that looks unrelated.